test/common/common/logger_test.cc

```diff
@@ -0,0 +1,47 @@
+#include <string>
+
+#include "common/common/logger.h"
+
+#include "gmock/gmock.h"
+#include "gtest/gtest.h"
+
+namespace Envoy {
+
+class LoggerEscapeTest : public testing::Test {};
+
+TEST_F(LoggerEscapeTest, LinuxEOL) {
+  const std::string log = Logger::DelegatingLogSink::escapeLogLine("line 1 \n line 2\n");
+  EXPECT_EQ("line 1 \\n line 2\n", log);
```

Comment:
nit: if you skip the temp variable in these tests, and a test fails for some reason on some platform (say there's one that doesn't love your usage of rend() in substraction) then the error messages will be incrementally clearer when tests fail. Also the tests will be more compact. E>g. for this one:
```
  EXPECT_EQ("line 1 \\n line 2\n", Logger::DelegatingLogSink::escapeLogLine("line 1 \n line 2\n"));
```
a failure here immediately shows the incorrect case without having to refer to a line number. This is a pretty trivial point though as the line-number would also be visible in the error log, so it would be unambiguous which testcase failed.

Reply:
Makes sense, done.